spark/src/main/scala/geotrellis/spark/timeseries/RDDTimeSeriesMethods.scala

```diff
@@ -0,0 +1,185 @@
+package geotrellis.spark.timeseries
+
+import geotrellis.raster._
+import geotrellis.raster.histogram._
+import geotrellis.raster.summary.polygonal._
+import geotrellis.spark._
+import geotrellis.spark.mask.Mask.Options
+import geotrellis.util.MethodExtensions
+import geotrellis.vector._
+
+import java.time.ZonedDateTime
+
+
+object RDDTimeSeriesFunctions {
+
+  def histogramProjection(tile: Tile): StreamingHistogram =
+    StreamingHistogram.fromTile(tile)
+
+  def histogramReduction(left: StreamingHistogram, right: StreamingHistogram): StreamingHistogram =
+    left + right
+
+  def meanReduction(left: MeanResult, right: MeanResult): MeanResult =
+    left + right
+
+  def maxReduction(left: Double, right: Double): Double =
+    scala.math.max(left, right)
+
+  def minReduction(left: Double, right: Double): Double =
+    scala.math.min(left, right)
+
+  def sumReduction(left: Double, right: Double): Double =
+    left + right
+}
+
+abstract class RDDTimeSeriesMethods
+    extends MethodExtensions[TileLayerRDD[SpaceTimeKey]] {
+
+  private def geometriesToMultiPolygons(
+    polygons: Traversable[Geometry]
+  ): Traversable[MultiPolygon] = {
+    polygons.map({ p =>
+      p match  {
+        case p: Polygon => MultiPolygon(p)
+        case p: MultiPolygon => p
+        case p: GeometryCollection => {
+          if (p.polygons.length > 0) MultiPolygon(p.polygons.head)
+          else if (p.multiPolygons.length > 0) p.multiPolygons.head
+          else throw new Exception(s"Cannot handle $p.")
+        }
+      }
+    })
+  }
+
+  def sumSeries(
+    polygon: Geometry,
+    options: Options
+  ): Map[ZonedDateTime, Double] =
+    sumSeries(List(polygon), options)
+
+  def sumSeries(
+    polygon: Geometry
+  ): Map[ZonedDateTime, Double] =
+    sumSeries(List(polygon), Options.DEFAULT)
+
+  def sumSeries(
+    polygons: Traversable[Geometry],
```

Comment:
Having this be Geometries makes the code less typesafe. If we pass in other geometries, we get a match error that is potentially not very descriptive. We should change this to only accept a Polygon or MultiPolygon, both versions (the user can create the MultiPolygon themselves if needed).